quinn-udp/src/unix.rs

```diff
@@ -506,7 +506,7 @@ fn recv(io: SockRef<'_>, bufs: &mut [IoSliceMut<'_>], meta: &mut [RecvMeta]) ->
 #[cfg(apple_fast)]
 fn recv(io: SockRef<'_>, bufs: &mut [IoSliceMut<'_>], meta: &mut [RecvMeta]) -> io::Result<usize> {
     let mut names = [MaybeUninit::<libc::sockaddr_storage>::uninit(); BATCH_SIZE];
-    let mut ctrls = [cmsg::Aligned(MaybeUninit::<[u8; CMSG_LEN]>::uninit()); BATCH_SIZE];
+    let mut ctrls = [cmsg::Aligned([0u8; CMSG_LEN]); BATCH_SIZE];
```

Comment:
With `CMSG_LEN` being `88` and `BATCH_SIZE` being `32`, this is not a small allocation. I will try to run a benchmark on a recent Mac to see whether it has a performance impact.

Reply:
`msghdr_x` below on line 510 is already 56 bytes and also an array of `BATCH_SIZE`. Not quite the same size but same ballpark.

Can we use `mem::zeroed` for the `ctrl`s array as well? Would that be faster?

Maybe we can leave it at `MaybeUninit` and only initialise up to `bufs.len()`?

Reply:
If I understand this code correctly, then `ctrls` already takes up 2kb of stack space, even with the use of `MaybeUninit`. So the only additional cost here is zeroing out the 2kb which I'd think isn't going to be that expensive.

Reply:
Would be good to do some basic benchmarking either way.

Reply:
`msghdr_x` has an extra field `msg_datalen` (compared to `msghdr`): https://github.com/apple/darwin-xnu/blob/2ff845c2e033bd0ff64b5b6aa6063a1f8f65aa32/bsd/sys/socket.h#L849

Pretty sure that isn't useful here, but wanted to mention it just in case.